### PR TITLE
CLI Publish Command & Scaffolding

### PR DESCRIPTION
**Pull request by crutchcorn**
This PR:

- [x] Introduces [Liftoff](https://github.com/gulpjs/liftoff) to support config files, including CJS & ESM
    - [x] Adds support for TS config files using esbuild
- [x] Introduces [Commander](https://github.com/tj/commander.js) to support specific CLI commands, help command, and more.
- [x] Exports ESM-only (no CJS support) programatic usage of the publish script
- [x] Adds a `"bin"` flag to `package.json` so `npx` and other scripts can call `tanstack-config` to run things
- [x] Writes initial docs

# Screenshots

image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,29 +11,32 @@
     "url": "https://github.com/sponsors/tannerlinsley"
   },
   "type": "module",
-  "types": "build/legacy/index.d.ts",
-  "main": "build/legacy/index.cjs",
-  "module": "build/legacy/index.js",
+  "types": "src/index.d.ts",
+  "module": "src/index.js",
```
Comment (lachlancollins) [inline package.json:15]: I would suggest we don't add this. It's not used by Node, and all modern bundlers (which we will be using) also read `"exports"`.

File: package.json [hunk 2]
```diff
@@ -11,29 +11,32 @@
...
   "exports": {
     ".": {
```
Comment (lachlancollins) [inline package.json:17]: Should we just use `"./publish"` instead?

**Review by lachlancollins — APPROVED** [review]
Oh this is _GOOD_. Just checking, would running `tanstack-config` without args work? Or do you need to pass in a `--config`? I also really like you can run the function without the CLI too!

**Comment by crutchcorn** [comment]
If you run `tanstack-config` without any args it will search the CWD of the command and find `tanstack.config.{js,ts}` and resolve that way

Just wanted to provide more than one option, especially since the tools I selected support it

**Pull request merged**